pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -222,9 +222,14 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	if !exists {
 		return nil
 	}
-	item, exists, err := pc.informer.GetStore().GetByKey(key)
-	if !exists || err != nil {
-		return nil
+	return pc.getPodByKey(key)
```

Comment:
Isn't getPodByIP fundamentally broken? I would expect this to be completely removed?

Reply:
Fundamentally broken as in we should index by a combo of network & address?

Reply:
Its not just network, in the "IP reach-ability" sense. If you are run as hostNetwork, the pod runs without its own dedicated pod network, and the IP of the node. For example:
```
kindnet-vhk6w                                1/1     Running   0          4h44m   172.17.0.3   kind-control-plane   <none>           <none>
kube-apiserver-kind-control-plane            1/1     Running   0          4h44m   172.17.0.3   kind-control-plane   <none>           <none>
kube-controller-manager-kind-control-plane   1/1     Running   0          4h44m   172.17.0.3   kind-control-plane   <none>           <none>
kube-proxy-kbdp9                             1/1     Running   0          4h44m   172.17.0.3   kind-control-plane   <none>           <none>
kube-scheduler-kind-control-plane            1/1     Running   0          4h44m   172.17.0.3   kind-control-plane   <none>           <none>
```

You can see they all have the same IP

Reply:
So this PR fixes one specific instance, but we still lookup other things by IP. This means, for example, we may see the wrong service account for an endpoint.

We also don't inject host network pods so most of this may not matter as much as it would otherwise

Reply:
I understand that, will file another commit. This is to make it  as smaller as possible